components/patina_smbios/src/manager/core.rs

```diff
@@ -455,7 +499,23 @@ impl SmbiosManager {
 
         let smbios_record = SmbiosRecord::new(record_header, producer_handle, data, string_count);
 
-        self.records.borrow_mut().push(smbios_record);
+        // Maintain invariant: Type 127 End-of-Table marker must always be last
+        let mut records = self.records.borrow_mut();
+
+        // Insert before Type 127 if it exists as the last element, otherwise append
+        if records.last().is_some_and(|r| r.header.record_type == 127) {
+            let last_index = records.len() - 1;
+            records.insert(last_index, smbios_record);
+        } else {
+            records.push(smbios_record);
+        }
```

Comment:
You could avoid the inspection and insert. The perf impact is basically zero, so current code is fine, but consider: 
```suggestion
        records.push(smbios_record);
        //Swap the inserted element with the one before it (if it exists) to ensure type 127 stays at the end.
        if (records.len() > 1) {
            records.swap(records.len()-1, records.len()-2);
        }
```